code/modules/gunse/modular_gun_parts.dm

```diff
@@ -426,6 +426,56 @@ ABSTRACT_TYPE(/obj/item/gun_parts/accessory)
 	icon_state = "it_revolver_short"
 	length = 13
 
+/obj/item/gun_parts/barrel/soviet/improvshort
+	name = "Improvised Short Barrel"
+	desc = "Some pipe shrouded with the remains of a soviet spacesuit"
+	spread_angle = -2
+	part_DRM = GUN_FOSS | GUN_SOVIET | GUN_ITALIAN
+	name_addition = "stubby"
+	length = 15
+	icon_state = "improvshortbarrel"
+	icon = 'icons/obj/items/modular_guns/mistfits.dmi'
+
+/obj/item/gun_parts/barrel/soviet/improvlong
+	name = "Improvised Long Barrel"
+	desc = "Several mismatched parts from old rifles, welded together just for you"
+	spread_angle = -4
+	part_DRM = GUN_FOSS | GUN_SOVIET | GUN_ITALIAN
+	name_addition = "rifled"
+	length = 53
```

Comment:
This is crazy powerful and has zero downsides. It's strictly 4 times better than the NT Special Long Barrel which is itself not intended for regular play but more of a test item. Will rebalance.

Reply:
I can make it so it can only go on soviet and foss guns. The whole premise behind the misfit is that it's a good gun if your *only* using "soviet" parts. As soon as you start putting other stuff on it, it gets bad quick. idk how rare the soviet stuff is gonna be yet though, nor really a good grasp on how lethality is gonna be yet.

Reply:
Changed to 35 length and added jaming. Stubby and Historical Barrels got changed to 13 and 25 respectively, with the Stubby barrel having 1 jam and the historical barrel having none